File: 1552-build-an-array-with-stack-operations/build-an-array-with-stack-operations.cs
public class Solution {
    public IList<string> BuildArray(int[] target, int n) {
        IList<string> ans = new List<string>();
        int j = 0;
        for (int i = 1; i <= n && j < target.Length; i++) {
            if (i == target[j]) {
                ans.Add("Push");
                j++;
            } else {
                ans.Add("Push");
                ans.Add("Pop");
            }
        }
        return ans;
    }
}

File: 1552-build-an-array-with-stack-operations/build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {

        string push ="Push";
        string pop="Pop";
        vector<string> ans;
        int j =0;
        for(int i=1;i<=n && j < target.size();i++){
            if(i==target[j]){
                ans.push_back(push);
                j++;
            }
            else{
                ans.push_back(push);
                 ans.push_back(pop);
            }
        }
        return ans;
    }
};